### PR TITLE
fix: incorrect switch case eval when server response code is 429

### PR DESCRIPTION
**Pull request by srikanthccv**
While working on tests for http sender I found that it doesn't respect the `retry-after` for too many requests response. I believe this is a bug.

### CODE REVIEW

File: client/internal/httpsender.go [hunk 1]
```diff
@@ -157,8 +157,7 @@ func (h *HTTPSender) sendRequestWithRetries(ctx context.Context) (*http.Response
 						h.callbacks.OnConnect()
 						return resp, nil
 
-					case http.StatusTooManyRequests:
-					case http.StatusServiceUnavailable:
+					case http.StatusTooManyRequests, http.StatusServiceUnavailable:
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply client/internal/httpsender.go:160]: Added [here](https://github.com/open-telemetry/opamp-go/pull/124/files#diff-15ece2a455633fee23cc3e9f092833e157845178723265fccc9167624eee9da0R53) along with other tests. It also proves that tests fail. Is that fine or do you want me to add test in this PR itself?

[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply client/internal/httpsender.go:160]: @srikanthccv please add the test in this PR.

**Review by tigrannajaryan — COMMENTED** [review]
Thanks for the catch.

File: client/internal/httpsender.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (srikanthccv) [reply client/internal/httpsender.go:165]: Added in 3a465c0c4daaff9dd2591621dc1fb2628c64444d.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**